File: tests/parsers/samples/code/kt/simple.kt
package samples

data class Metrics(val baseline: Double, val rms: Double)

fun runSimple(): Metrics {
    val analyzer = ComplexMetrics.buildDefault()
    val rms = analyzer.rootMeanSquare(listOf(1.0, 2.0, 3.0))
    val baseline = analyzer.distance(10, 3)
    return Metrics(baseline, rms)
}

fun summarize(): String {
    val metrics = runSimple()
    return "baseline=${metrics.baseline}, rms=${String.format("%.2f", metrics.rms)}"
}

File: tests/parsers/samples/code/php/simple.php
<?php

namespace Samples;

use Samples\ComplexMetrics;

function run_simple(): array
{
    $analyzer = ComplexMetrics::buildDefault();
    return [
        'baseline' => $analyzer->distance(10, 3),
        'rms' => $analyzer->rootMeanSquare([1.0, 2.0, 3.0]),
    ];
}

function summarize(): string
{
    $metrics = run_simple();
    return sprintf('baseline=%.2f, rms=%.2f', $metrics['baseline'], $metrics['rms']);
}
